framework/auth/views.py

```diff
@@ -108,14 +108,16 @@ def forgot_password_get(auth, *args, **kwargs):
 # Log in
 ###############################################################################
 
-# TODO: Rewrite async
 @collect_auth
 def auth_login(auth, **kwargs):
     """If GET request, show login page. If POST, attempt to log user in if
     login form passsed; else send forgot password email.
 
     """
+    campaign = request.args.get('c')
```

Comment:
Thanks- doing it now makes the most sense. We can change a lot of things after release, but we should try not to break anything user-facing (like URLs) once they're pushed out to the world.